fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/NormalizePredicateRule.java

```diff
@@ -124,4 +126,45 @@ public ScalarOperator visitCompoundPredicate(CompoundPredicateOperator predicate
 
         return predicate;
     }
+
+    /**
+     * Rewrite column ref into comparison predicate *
+     * Before
+     * example:
+     * IN
+     * / |  \
+     * left 1  a  b
+     * After rule:
+     * left = 1 OR left = a OR left = b
+     */
+    @Override
+    public ScalarOperator visitInPredicate(InPredicateOperator predicate, ScalarOperatorRewriteContext context) {
+        List<ScalarOperator> rhs = predicate.getChildren().subList(1, predicate.getChildren().size());
+        if (rhs.stream().allMatch(ScalarOperator::isConstant)) {
+            return predicate;
+        }
+
+        List<ScalarOperator> result = new ArrayList<>();
+        ScalarOperator lhs = predicate.getChild(0);
+        boolean isIn = !predicate.isNotIn();
+
+        for (ScalarOperator child : predicate.getChildren().subList(1, predicate.getChildren().size())) {
+            BinaryPredicateOperator newOp;
+            if (isIn) {
+                newOp = new BinaryPredicateOperator(BinaryPredicateOperator.BinaryType.EQ, lhs, child);
+            } else {
+                newOp = new BinaryPredicateOperator(BinaryPredicateOperator.BinaryType.NE, lhs, child);
+            }
+            result.add(newOp);
+        }
+        if (result.size() == 1) {
+            return result.get(0);
+        }
```

Comment:
```suggestion
        if (result.size() == 1) {
            return result.get(0);
        }
```
Don't need this if you use `Utils.compoundxx`